esphome/components/oxt_dimmer/oxt_dimmer.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+/*
+  Copyright © 2023
+*/
+
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/component.h"
+#include "esphome/components/uart/uart.h"
+#include "esphome/components/light/light_output.h"
+#include "esphome/components/light/light_state.h"
+#include "esphome/components/light/light_traits.h"
+
+namespace esphome {
+namespace oxt_dimmer {
+
+class OxtController;
+
+/**
+ * OxtDimmerChannel inherits from light::LightOutput and provides "light"
+ * functionality towards front-end, ESPHome, HASS...
+ */
+class OxtDimmerChannel : public light::LightOutput, public Component {
+ public:
+  // Component overrides
+  void dump_config() override;
+
+  // LightOutput overrides
+  light::LightTraits get_traits() override;
+  void setup_state(light::LightState *state) override { light_state_ = state; }
```

Comment:
> All uses of class members and member functions should be prefixed with this-> to distinguish them from global functions/variables.

https://developers.esphome.io/contributing/code/#c-code-style
```suggestion
  void setup_state(light::LightState *state) override { this->light_state_ = state; }
```